kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md

```diff
@@ -191,7 +191,7 @@ module EVM-OPTIMIZATIONS
       ...
     </kevm>
     requires ( #if USEGAS #then Gverylow < SCHED > <=Gas GAVAIL #else true #fi )
-     andBool ( #sizeWordStack( chop( ( W0 +Int W1 ) ) : WS ) <=Int 1024 )
+     andBool ( #sizeWordStack( WS ) <=Int 1023 )
```

Comment:
If `#sizeWordStack( WS ) ==Int 1023`, then `#sizeWordStack( W0 : W1 : WS ) ==Int 1025`, which is malformed. I believe that it should be: 
```suggestion
     andBool ( #sizeWordStack( WS ) <=Int 1022 )
```

Reply:
That's not the way the rule is summarized. There post-execution stack check (for overflow) checks that _after_ the needed stack elements have been removed, and the new ones are added, that there isn't a stack overflow. It does not check that the stack _going into_ the operation is overflowing.

These summaries were auto-generated by executing each opcode symbolically directly, which is why they are exactly the expressions that showed up. Putting the tighter check on the stack would be a change to the semantics (a change that is still true, as CI tells us, but is more restrictive).

Reply:
The point here is to partially evaluate _exactly_ the expressions that are collected during symbolic execution, which is what happens here.

Reply:
Very well.